src/algorithms/reco/FarForwardNeutronReconstruction.cc

```diff
@@ -1,26 +1,37 @@
 // SPDX-License-Identifier: LGPL-3.0-or-later
-// Copyright (C) 2024 Sebouh Pual
+// Copyright (C) 2024 Sebouh Paul
 #include "FarForwardNeutronReconstruction.h"
-
+#include <JANA/JException.h>
 #include <edm4eic/ClusterCollection.h>
 #include <edm4eic/ReconstructedParticleCollection.h>
 #include <edm4hep/Vector3f.h>
 #include <math.h>
 #include <gsl/pointers>
+#include <vector>
 
 /**
- Creates a "neutron candidate" Reconstructed Particle consisting of all clusters in a given ClusterCollection.
- Its energy is the sum of the energies of the constituent clusters, and its direction is the direction from the
-origin to the position of the most energetic cluster
+ Creates a "neutron candidate" Reconstructed Particle consisting of all clusters in a
+ given ClusterCollection.  Its energy is the sum of the energies of the constituent clusters
+ times a correction factor, and its direction is the direction from the origin to the position
+ of the most energetic cluster.  The correction factor is given by 1/(1+c[0]+c[1]/sqrt(E)+c[2]/E),
+ where c is the coefficients and E is the uncorrected energy in GeV.  This form was chosen
+ empirically based on the discrepancies in single-neutron MC simulations between the uncorrected
+ reconstructed energies and the truth energies of the neutrons.
  */
 
 namespace eicrecon {
 
-    void FarForwardNeutronReconstruction::init() {  }
-
+    void FarForwardNeutronReconstruction::init() {
+      if (m_cfg.scale_corr_coeff.size() < 3) {
+        throw JException("Invalid configuration.  m_cfg.scale_corr_coeff should have at least 3 parameters");
```

Comment:
IIRC, for now we still need to print the message, also we can't use JANA anything in algorithms.
```suggestion
        error("Invalid configuration.  m_cfg.scale_corr_coeff should have at least 3 parameters");
        throw std::runtime_error("Invalid configuration.  m_cfg.scale_corr_coeff should have at least 3 parameters");
```